src/Mjarestad/Filtry/Filtry.php

```diff
@@ -510,14 +522,17 @@ protected function filterWalker($filter, $data)
 	 * @param  array $data
 	 * @return string|array
 	 */
-	protected function filterExtensionWalker($filter, $data)
+	protected function filterExtensionWalker($filter, $data, $parameters = [])
 	{
+		array_unshift($parameters, $data);
+
 		if (is_array($data) and $this->recursive === true) {
 			foreach ($data as $key => $value) {
-				$data[$key] = call_user_func($this->extensions[$filter], $data);
```

Comment:
This PR also fix a bad issue where recursive doesn't work with custom filters (extensions). The `call_user_func` function is call with `$data` instead of `$value` inside the `foreach`.